vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt

```diff
@@ -74,9 +76,16 @@ class TchapRegisterFragment @Inject constructor(private val platformViewModelFac
             }.exhaustive
         }
 
-        loginViewModel.observeViewEvents {
-            when (it) {
+        loginViewModel.observeViewEvents { loginViewEvents ->
+            when (loginViewEvents) {
                 TchapLoginViewEvents.OnLoginFlowRetrieved -> loginViewModel.handle(TchapLoginAction.LoginOrRegister(login, password, getString(R.string.login_default_session_public_name)))
+                is TchapLoginViewEvents.RegistrationFlowResult -> {
+                    val emailStage = loginViewEvents.flowResult.missingStages.firstOrNull { it.mandatory && it is Stage.Email }
+
+                    if (emailStage != null) {
+                        loginViewModel.handle(TchapLoginAction.AddThreePid(RegisterThreePid.Email(login)))
```

Comment:
I would prefer to check whether login is actually an email address before using it
(Do you know whether element-android check email string before using it)